### PR TITLE
[MM-48947] app/platform: do not restart metrics on every config save

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
It seems like there are no checks if a metrics server restart is required or not.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48947

#### Release Note

```release-note
Bugfix on metrics server restart when there is a config change.
```

### CODE REVIEW

File: app/platform/service_test.go [hunk 1]
```diff
@@ -117,17 +116,14 @@ func TestMetrics(t *testing.T) {
...
-		th.Service.SaveConfig(th.Service.Config(), false)
+		cfg.MetricsSettings.Enable = model.NewBool(false)
+		th.Service.SaveConfig(cfg, false)
 
 		_, err = http.Get(metricsAddr)
 		require.Error(t, err)
 	})
 
 	t.Run("ensure the metrics server is started with advanced metrics", func(t *testing.T) {
-		t.Skip("MM-47635")
```
Comment (isacikgoz) [inline app/platform/service_test.go:130]: This must be creep in by https://github.com/mattermost/mattermost-server/pull/21401

**Review by agarciamontoro — APPROVED** [review]
Neat, thanks!

**Comment by isacikgoz** [comment]
@agnivade requesting a review again as I removed the trigger from `SaveConfig` and added to config listeners. Because there are cases where the config is getting updated with `UpdateConfig` method and this trigger was getting ignored.

**Review by agnivade — APPROVED** [review]

**Pull request merged**